src/app/qgsdatasourcemanagerdialog.h

```diff
@@ -0,0 +1,74 @@
+/***************************************************************************
+    qgsdatasourcemanagerdialog.h - datasource manager dialog
+
+    ---------------------
+    begin                : May 19, 2017
+    copyright            : (C) 2017 by Alessandro Pasotti
+    email                : apasotti at itopen dot it
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSDATASOURCEMANAGERDIALOG_H
+#define QGSDATASOURCEMANAGERDIALOG_H
+
+#include <QList>
+#include <QDialog>
+#include "qgsoptionsdialogbase.h"
+#include "qgsguiutils.h"
+
+class QgsBrowserDockWidget;
+class QgsRasterLayer;
+class QgsMapCanvas;
+
+namespace Ui
+{
+  class QgsDataSourceManagerDialog;
+}
+
+class QgsDataSourceManagerDialog : public QgsOptionsDialogBase
```

Comment:
@nyalldawson @wonder-sk  now that I've moved the browser to `gui` I'll probably be able to work on moving the new layers dialog to `gui` in the next weeks. One problem I see is that there is a dependency on `QgsCanvas` for the ArcGIS provider that needs to be solved, I asked @manisandro about that but I've got no answer. 
https://github.com/qgis/QGIS/blob/master/src/app/qgsdatasourcemanagerdialog.cpp#L112
I'm not familiar with that provider, do you have any hint?